tests/Type/SimpleTypeTest.php

```diff
@@ -9,32 +9,36 @@
 use Consistence\Sentry\Metadata\SentryIdentificator;
 use Consistence\Sentry\Metadata\SentryMethod;
 use Consistence\Sentry\Metadata\Visibility;
+use Generator;
 use PHPUnit\Framework\Assert;
 
 class SimpleTypeTest extends \PHPUnit\Framework\TestCase
 {
 
-	public function testGenerateGet(): void
+	/**
+	 * @return mixed[][]|\Generator
+	 */
+	public function generateMethodDataProvider(): Generator
 	{
-		$integerSentry = new SimpleType();
-		$getMethod = new SentryMethod(
-			new SentryAccess('get'),
-			'getFoo',
-			Visibility::get(Visibility::VISIBILITY_PUBLIC)
-		);
-		$propertyMetadata = new PropertyMetadata(
-			'fooProperty',
-			FooClass::class,
-			'int',
-			new SentryIdentificator('int'),
-			false,
-			[
-				$getMethod,
-			],
-			null
-		);
-
-		$method = '
+		yield 'get' => (function (): array {
+			$sentryMethod = new SentryMethod(
+				new SentryAccess('get'),
+				'getFoo',
+				Visibility::get(Visibility::VISIBILITY_PUBLIC)
+			);
+			$propertyMetadata = new PropertyMetadata(
```

Comment:
I think the variables `$propertyMetadata` and `$expectedMethod` are not needed?

Reply:
Removed.